File: Amazon/3_quora.py



def removeAlternateDuplicates(myStr):
	myStr = list(myStr)

	letters = [0] * 256
	ind = 0
	for i in range(len(myStr)):
		lowerCase = myStr[i] # char
		if(ord(myStr[i]) >= 65 and ord(myStr[i]) <= 90 ):
			# lowerCase += ord('a') - ord('A')
			lowerCase.lower()
		if (letters[ord(lowerCase)] == 0):
			letters[ord(lowerCase)] += 1
			myStr[ind] = myStr[i]
			ind += 1
		else:
			letters[ord(lowerCase)] -= 1


	while(ind < len(myStr)):
		myStr[ind] = ""
		ind += 1

	return ''.join(myStr)

def main():
	myStr = "aAaBbBcCcdefgFGZzzzO"
	print removeAlternateDuplicates(myStr)


if __name__ == "__main__":
	main()

File: Amazon/3_quora.cc
#include<cstdio>



void removeAlternateDuplicates(char str[],int n){
	int letters[256];
	int i = 0;
	int index=0;
	for(i=0;i<256;i++)
		letters[i]=0;
	for(i=0; i<n;i++){
		char lowerCase = str[i];
		if(str[i] >= 65 && str[i] <= 90){
			lowerCase +=('a' - 'A');
		}
		if(letters[lowerCase] == 0){
			letters[lowerCase]++;
			str[index]=str[i];
			index++;
		}else{
			letters[lowerCase]--; // We want to Alternate
		}
	}
	while(index < n){
		str[index]= 0;
		index++;
	}
}
int main(){
	char str[]="aAaBbBcCcdefgFGZzzzO"; //aaBBccdefgZzO
	// char str[]="you got beautiful eyes"; 
	// char str[]="Today is the Day"; // Today ishe
	removeAlternateDuplicates(str,20);
	printf("%s\n",str );
	printf("%s\n","Kaisy hoo" );
	return 0;


}